script.js

```diff
@@ -1,4 +1,343 @@
+    input != 2 &&
+    input != `two` &&
+    input != 3 &&
+    input != `three` &&
+    input != 4 &&
+    input != `four` &&
+    input != 5 &&
+    input != `five`
+  ) {
+    console.log(`gameState`, gameState);
+    return `Sigh. Are my instructions difficult? Come on now, how many dice do you wish to roll for each player? Choose a number from 2 - 5.`;
+  }
+
+  if (
+    gameState == `gameSettingsNumDice` &&
+    (input == 2 ||
+      input == `two` ||
+      input == 3 ||
+      input == `three` ||
+      input == 4 ||
+      input == `four` ||
+      input == 5 ||
+      input == `five`)
+  ) {
+    numberOfDice = input;
+    gameState = `setNumberOfRounds`;
+    console.log(`gameState`, gameState);
+    console.log(`numberOfDice`, numberOfDice);
+
+    return `Okay, so we will be using ${input} dice for each round.<br><br>
+    Now one last thing to figure out -- How many rounds will you be playing? Choose from 2 to 5 rounds. I will entertain you but I haven't got all day!`;
+  }
+
+  if (
+    gameState == `setNumberOfRounds` &&
+    input != 2 &&
+    input != `two` &&
+    input != 3 &&
+    input != `three` &&
+    input != 4 &&
+    input != `four` &&
+    input != 5 &&
+    input != `five`
+  ) {
+    console.log(`gameState`, gameState);
+    return `Listen, we're nearly there. All you have to do is just decide how many rounds you're going to subject yourself to. Just decide. Anything between <b>2 to 5.</b>`;
+  }
+
+  if (
+    gameState == `setNumberOfRounds` &&
+    (input == 2 ||
+      input == `two` ||
+      input == 3 ||
+      input == `three` ||
+      input == 4 ||
+      input == `four` ||
+      input == 5 ||
+      input == `five`)
+  ) {
+    numberOfRounds = input;
+    gameState = `readyToPlay`;
+    console.log(`gameState`, gameState);
+    console.log(`numberOfRounds`, numberOfRounds);
+
+    return `Wow. I never thought we'd get here but looks like we finally got all we need to begin the game.<br>
+    <br>
+    Looks like you'll be playing ${input} rounds under my watchful eye.<br>
+    <br>
+    Now, <i>take a deep breath</i> and <b>speak to me</b> when you're ready for fate to deal everyone's hands and I'll reveal the secrets of the chamber to you.`;
+  }
+
+  //Gameplay code begins here
+
+  if (gameState == `readyToPlay`) {
+    console.log(`gameState`, gameState);
+    console.log(`roundsPlayed`, roundsPlayed);
+
+    while (roundsPlayed < numberOfRounds) {
+      {
+        playersSubmissions = [];
+        console.log(`current round`, roundsPlayed + 1);
+        for (
+          var totalPlayersRollCount = 0;
+          totalPlayersRollCount < numberOfPlayers;
+          totalPlayersRollCount += 1
+        ) {
+          playersDiceRolls = [];
+          for (
+            var diceRollCount = 0;
+            diceRollCount < numberOfDice;
+            diceRollCount += 1
+          ) {
+            var diceRoll = function () {
+              return Math.ceil(Math.random() * 6);
+            };
```

Comment:
i like that you shortened your diceRoll to something so short! but maybe move this out of the main function (functions declared before main )